File: global/501000.lua

local config_loans = dofile("/home/eqemu/server/quests/global/loan_config.lua")

-- Supermob depop timer: configurable from loan_config.lua
function event_spawn(e)
	local depop_ms = (config_loans.SUPER_MOB_DEPOP_MINUTES or 3) * 60000 -- minutes to ms
	eq.set_timer("depop", depop_ms)
end

function event_timer(e)
	if e.timer == "depop" then
		eq.stop_timer("depop")
		eq.debug("Depop timer fired. Attempting to depop all supermobs with ID: " .. tostring(config_loans.SUPER_MOB_NPC_ID))
		eq.depop_all(config_loans.SUPER_MOB_NPC_ID)
	end
end

-- Despawn supermob if it kills a player
function event_killed(e)
	if e.killed:IsClient() then
		eq.debug("Supermob killed a player. Attempting to depop all supermobs with ID: " .. tostring(config_loans.SUPER_MOB_NPC_ID))
		eq.depop_all(config_loans.SUPER_MOB_NPC_ID)
	end
end

File: global/501000.pl
use lib '.';
my $config_loans = plugin::loan_config();

# EVENT_SPAWN: Set depop timer using config
sub EVENT_SPAWN {
    my $depop_ms = ($config_loans->{SUPER_MOB_DEPOP_MINUTES} // 3) * 60 * 1000;
    quest::settimer('depop', $depop_ms);
}

# EVENT_TIMER: Depop all supermobs when timer fires
sub EVENT_TIMER {
    my ($timer_name) = @_;
    if ($timer_name eq 'depop') {
        quest::stoptimer('depop');
        quest::debug("Depop timer fired. Attempting to depop all supermobs with ID: $config_loans->{SUPER_MOB_NPC_ID}");
        quest::depopall($config_loans->{SUPER_MOB_NPC_ID});
    }
}

# EVENT_KILLED: Depop all supermobs if this NPC kills a player
sub EVENT_KILLED {
    my ($killed_entity) = @_;
    if ($killed_entity && $killed_entity->IsClient()) {
        quest::debug("Supermob killed a player. Attempting to depop all supermobs with ID: $config_loans->{SUPER_MOB_NPC_ID}");
        quest::depopall($config_loans->{SUPER_MOB_NPC_ID});
    }
}